nvd/nvd.go

```diff
@@ -99,11 +100,16 @@ func (u Updater) Update() error {
 			}
 			slog.Info("Fetched NVD entries", slog.Int("total", totalResults), slog.Int("start_index", startIndex))
 		}
+		totalEntries += totalResults
 	}
 
-	// Update last_updated.json at the end.
-	if err = utils.SetLastUpdatedDate(apiDir, u.lastModEndDate); err != nil {
-		return xerrors.Errorf("unable to update last_updated.json file: %w", err)
+	// If NVD didn't return records for all intervals
+	// we shouldn't update the last update as it might be a bug.
+	if totalEntries > 0 {
```

Comment:
When I was writing the PR description, I thought we should put the error back here so we could see it right away. But maybe that's too mach.